execute/allocator.go

```diff
@@ -62,6 +63,14 @@ func (a *Allocator) GrowBools(slice []bool, n int) []bool {
 	return s
 }
 
+// SliceBools reduces the input slice to slice[start:stop]
+func (a *Allocator) SliceBools(slice []bool, start, stop int) []bool {
+	origLen := len(slice)
+	slice = slice[start:stop]
+	a.Free(origLen-len(slice), boolSize)
```

Comment:
only reason I did this was because that's how all other changes are made to the ColListTable columns.  But you are right about the backing array.  This call to Free is probably not safe.  

We could copy out the data and discard the backing array, what do you think of that?